feat Nextcloudclient

gg46ixav opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Deploy now supports three modes: Classic, Metadata-file (JSON), and Nextcloud WebDAV upload.
  * Nextcloud mode uploads files/directories, computes per-file SHA‑256 checksums and sizes, and returns remote URLs for deployment.

* **Improvements**
  * New CLI flags for mode-driven workflows with validation to prevent conflicting options.
  * README updated with mode-specific usage, examples, metadata format, and authentication notes.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

gg46ixav commented:
To review:
- changed the dataset context url in databusclient/client.py
- added "Deploy to Databus" and "Upload to Nextcloud and Deploy to Databus" to README.md

Integer-Ctrl commented on README.md:
````diff
@@ -64,6 +64,44 @@ docker run --rm -v $(pwd):/data dbpedia/databus-python-client download https://d
+Then registers them on the databus.
+### Example Call
+```bash
+python -m databusclient.deploy \
````
How should this be used? There is the cli command `databusclient --deploy` but you created a new method which also reads args.

Integer-Ctrl commented on databusclient/metadata.json:
What this file should do is not very well described. So if `--no-upload` is used, it needs to get filename, size, download uri and sha256?

gg46ixav commented:
I removed the deploy.py and added a upload_and_deploy and a deploy_with_metadata function to cli.py.
Furthermore I refactored some code and added help functions for these two functions to client.py.
Matching those changes I updated the README.md.
I also changed the context url back to the initial one.

gg46ixav commented:
I added a very simple validate_distributions function in client.py to validate the given distributions when calling the classic deploy command.

Integer-Ctrl commented:
> I added a very simple validate_distributions function in client.py to validate the given distributions when calling the classic deploy command.

Not needed.

> I can add some tests. Are there specific things I definitely need to test? Otherwise I would implement what I think is useful/appropriate. And should I use a new branch for that?

I don't have time right now to think about meaningful tests. It's not a must-have, but if you want to work on it, doing so in a new branch would be best. (Tests are important, but the whole project isn't well tested anyway..)

If there’s anything else you’d like to change, please comment. Otherwise, if your **local tests of deploy work fine**, just let me know and I’ll merge the PR.
